lib/tapioca/compilers/dsl/sequel_columns.rb

```diff
@@ -0,0 +1,213 @@
+      #   def title=(value); end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def title?(*args); end
+      # end
+      # ~~~
+      class SequelColumns < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(Sequel::Model)).void }
+        def decorate(root, constant)
+          return if constant.table_name.blank?
+          return if constant.name.blank?
+
+          module_name = "#{constant}::GeneratedAttributeMethods"
+          root.create_module(module_name) do |mod|
+            constant.columns.each do |column_name|
+              column_name = column_name.to_s
+              add_methods_for_attribute(mod, constant, column_name)
+            end
+          end
+
+          root.path(constant) do |klass|
+            klass.create_include(module_name)
+          end
+        end
+
+        sig { override.returns(T::Enumerable[Module]) }
+        def gather_constants
+          Sequel::Model.descendants
+        end
+
+        private
+
+        sig do
+          params(
+            klass: Parlour::RbiGenerator::Namespace,
+            name: String,
+            methods_to_add: T.nilable(T::Array[String]),
+            return_type: T.nilable(String),
+            parameters: T::Array[[String, String]]
+          ).void
+        end
+        def add_method(klass, name, methods_to_add, return_type: nil, parameters: [])
+          create_method(
+            klass,
+            name,
+            parameters: parameters.map do |param, type|
+              Parlour::RbiGenerator::Parameter.new(param, type: type)
+            end,
+            return_type: return_type
+          ) if methods_to_add.nil? || methods_to_add.include?(name)
+        end
+
+        sig do
+          params(
+            klass: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(Sequel::Model),
+            column_name: String,
+            attribute_name: String,
+            methods_to_add: T.nilable(T::Array[String])
+          ).void
+        end
+        def add_methods_for_attribute(klass, constant, column_name, attribute_name = column_name, methods_to_add = nil)
+          getter_type, setter_type = type_for(constant, column_name)
+
+          add_method(
+            klass,
+            attribute_name.to_s,
+            methods_to_add,
+            return_type: getter_type
+          )
+
+          add_method(
+            klass,
+            "#{attribute_name}=",
+            methods_to_add,
+            parameters: [["value", setter_type]],
+            return_type: setter_type
+          )
+        end
+
+        sig do
+          params(
+            constant: T.class_of(Sequel::Model),
+            column_name: String
+          ).returns([String, String])
+        end
+        def type_for(constant, column_name)
+          return ["T.untyped", "T.untyped"] if do_not_generate_strong_types?(constant)
```

Comment:
Do you also need the `StrongTypeGeneration` pattern for sequel or is it to mirror what we do in the Active Record generators?

We might want the reverse choice, something like `UnsafeTypeGeneration` as to encourage safe typing by default.